src/helpers/script_helpers.sh

```diff
@@ -14,6 +14,7 @@ function command_steps {
   echo_if_all_true "change_directory $(git_root)" "$RUN_IN_GIT_ROOT" "$IN_SUB_FOLDER"
   echo_if_all_true "stash_open_changes" "$STASH_OPEN_CHANGES" "$INITIAL_OPEN_CHANGES"
   steps
+  echo_if_all_true "set_previous_branch \$(expected_previous_branch '$INITIAL_PREVIOUS_BRANCH_NAME' '$INITIAL_BRANCH_NAME')"
```

Comment:
What do you think about a method `preserve_checkout_history`
Currently it would just be
```sh
function preserve_checkout_history {
  local initial_previous_branch_name=$1
  local initial_branch_name=$2
  local desired_previous_branch=$(expected_previous_branch "$initial_previous_branch_name" "$initial_branch_name")
  set_previous_branch "$desired_pervious_branch"
}
```
Then here could just become
```
"preserve_checkout_history $INITIAL_PREVIOUS_BRANCH_NAME $INITIAL_BRANCH_NAME"
```
I think that will be more flexible in terms of undoing things